src/ncp/ncp_base_mtd.cpp

```diff
@@ -4552,6 +4552,35 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_PHY_PCAP_ENABLED>(voi
     return error;
 }
 
+template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_NET_LEAVE_GRACEFULLY>(void) { return OT_ERROR_NONE; }
+
+template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_NET_LEAVE_GRACEFULLY>(void)
+{
+    otError error = OT_ERROR_NONE;
+    bool    leave;
+
+    SuccessOrExit(error = mDecoder.ReadBool(leave));
+
+    error = otThreadDetachGracefully(mInstance, ThreadDetachGracefullyHandler, this);
+
+exit:
+    return error;
+}
+
+void NcpBase::ThreadDetachGracefullyHandler(void *aContext)
+{
+    static_cast<NcpBase *>(aContext)->ThreadDetachGracefullyHandler();
+}
+
+void NcpBase::ThreadDetachGracefullyHandler(void)
+{
+    spinel_status_t status = SPINEL_STATUS_THREAD_DETACH_GRACEFULLY_DONE;
+
+    IgnoreError(otInstanceErasePersistentInfo(mInstance));
```

Comment:
Do we need this call here? Should it be done by OT core if needed?

Reply:
It seems that calling `otThreadDetachGracefully` doesn't trigger `otInstanceErasePersistentInfo`. And OT core doesn't call `otInstanceErasePersistentInfo` or `Instance::ErasePersistentInfo` either.

Reply:
I don't believe "graceful detach" is intended to earse non-volatile settings (act like a "factory reset").

Its purpose is for a device to leave the network gracefully (informing others it's leaving). One use case is when a device knows it will soon undergo a software update and wants to rejoin the same network afterward.

Reply:
That makes sense 👍  I've removed it. We can let host send a `SPINEL_CMD_NET_CLEAR` to trigger the erase when there is a need.